docs: add java client set up guide (#319)
abcxyz/lumberjack: docs/clients.md

## Java

> It is not recommended and supported to "assemble" a client in code,

Reviewer: ```suggestion
Currently we don't support "assembling" a client in Java code. Please file an issue if you require such capability.
```

Reviewer: And move to the other subsection

> please use config file. The default config path is `audit_logging.yml`, or
> if it is overwritten via env var, AUDIT_CLIENT_CONFIG_PATH.

Reviewer: ```suggestion
The default config path is `audit_logging.yml`. You can override it with env var `AUDIT_CLIENT_CONFIG_PATH`.
```

Reviewer: Move this to "### Create a client from a config file"

### Create a client from a config file

> ```java

Reviewer: Should we provide a intro to Guice, e.g. a link?

Injector injector = Guice.createInjector(new AuditLoggingModule());

// For gRPC services, use AuditLoggingServerInterceptor.
AuditLoggingServerInterceptor interceptor =
        injector.getInstance(AuditLoggingServerInterceptor.class);

// Otherwise, use LoggingClient directly.
LoggingClient client = injector.getInstance(LoggingClient.class);
```

> ### Extend

Reviewer: Let's have the same subsections as Go. But under "### "Assemble" a client in code" we say what you say at the top: "It is not supported..."

> Extending with custom processors is not supported at the moment, please file an issue if needed.

Reviewer: ```suggestion
Due to all the log processors are initialized via injection, we currently don't support vending custom log processors to the audit log client. Please file an issue if you need such capability.
```
